### PR TITLE
Internal: Site planner generated website tagging ability [ED-16361]

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: modules/ai/module.php [hunk 1]
```diff
@@ -165,6 +167,41 @@ public function __construct() {
...
+			return;
+		}
+
+		if ( 'site-settings' !== $imported_data['runner'] ) {
+			return;
+		}
+
+		$is_connected = $this->get_ai_app()->is_connected() && User::get_introduction_meta( 'ai_get_started' );
+
+		if ( ! Tracker::is_allow_track() && ! $is_connected ) {
```
Comment (matipojo) [inline modules/ai/module.php:181]:
```suggestion
		if ( ! $is_connected ) {
```

File: modules/ai/module.php [hunk 2]
```diff
@@ -165,6 +167,41 @@ public function __construct() {
 		add_action( 'elementor/element/common/_section_transform/after_section_end', [ $this, 'register_ai_hover_effect_control' ], 10, 1 );
 	}
 
+	public function handle_kit_install( $imported_data ) {
+		if ( 'success' !== $imported_data['status'] ) {
```
Comment (louiswol94) [inline modules/ai/module.php:171]:
```suggestion
		if ( isset( $imported_data['status'] ) && 'success' !== $imported_data['status'] ) {
```
Maybe, just to be super safe?

File: modules/ai/connect/ai.php [hunk 3]
```diff
@@ -84,6 +84,32 @@ public function get_remote_frontend_config( $data ) {
...
+	 *     @type array $data
+	 *     @type array $client {
+	 *         @type string $name
+	 *         @type string $version
+	 *         @type string $session_id
+	 *     }
+	 * }
+	 */
+	public function send_event( array $event_data ) : void {
+		$this->ai_request(
```
Comment (louiswol94) [inline modules/ai/connect/ai.php:99]: Can this block the editor in case the service goes down?

**Review by louiswol94 — DISMISSED** [review]

File: modules/ai/connect/ai.php (hunk 3 above)
Comment (matipojo) [reply modules/ai/connect/ai.php:99]: No, it happens after the import kit settings Ajax call, so the editor will work as usual.

**Review by louiswol94 — DISMISSED** [review]

File: modules/ai/module.php [hunk 4]
```diff
@@ -165,6 +167,41 @@ public function __construct() {
 		add_action( 'elementor/element/common/_section_transform/after_section_end', [ $this, 'register_ai_hover_effect_control' ], 10, 1 );
 	}
 
+	public function handle_kit_install( $imported_data ) {
+		if ( isset( $imported_data['status'] ) && 'success' !== $imported_data['status'] ) {
+			return;
+		}
+
+		if ( 'site-settings' !== $imported_data['runner'] ) {
```
Comment (louiswol94) [inline modules/ai/module.php:175]:
```suggestion
		if ( isset( $imported_data['runner'] ) && 'site-settings' !== $imported_data['runner'] ) {
```

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**